src/compiler/checker.ts

```diff
@@ -45073,6 +45087,44 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return value;
     }
 
+    function evaluatesToNumericLiteral(expr: Expression): boolean {

```

Comment:
Nit: I think this name describes an oxymoron. Maybe `isSyntacticallyNumeric`?

Reply:
Done.